src/track/cue.cpp

```diff
@@ -85,6 +111,24 @@ Cue::Cue(TrackId trackId, mixxx::AudioSignal::SampleRate sampleRate, const mixxx
     }
 }
 
+void Cue::readInfo(
+        mixxx::AudioSignal::SampleRate sampleRate,
+        mixxx::CueInfo* pCueInfo,
+        bool* pDirty) const {
+    DEBUG_ASSERT(pCueInfo);
+    QMutexLocker lock(&m_mutex);
+    *pCueInfo = mixxx::CueInfo(
```

Comment:
Why not return a `CueInfo` object (or the `m_Dirty` boolean if you pref) instead of writing to a pointer?

Reply:
I just copied the pattern that is currently used for track objects. Using an _out parameter_ instead of copying/moving big structures with many fields is more efficient at some point. Here it might not matter. We can't return a references because of the locking.